File: app/src/main/java/kei/magnet/classes/Location.java
package kei.magnet.classes;

import android.os.Bundle;
import android.os.Parcel;
import android.os.Parcelable;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by Dimitri on 27/10/2015.
 */
public class Location implements Parcelable{
    private double latitude;
    private double longitude;

    public Location(JSONObject jsonObject){
        try {
            this.latitude = jsonObject.getDouble("latitude");
            this.longitude = jsonObject.getDouble("longitude");
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    public Location(double latitude, double longitude){
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    // PARCELABLE
    @Override
    /**
     * Describe the kinds of special objects contained in this Parcelable's
     * marshalled representation.
     *
     * @return a bitmask indicating the set of special object types marshalled
     * by the Parcelable.
     */
    public int describeContents() {
        return 0;
    }

    /**
     * Flatten this object in to a Parcel.
     *
     * @param dest  The Parcel in which the object should be written.
     * @param flags Additional flags about how the object should be written.
     *              May be 0 or {@link #PARCELABLE_WRITE_RETURN_VALUE}.
     */
    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeDouble(latitude);
        dest.writeDouble(longitude);
    }

    /**
     * Instanciate a game using Parcelable
     * @param in
     */
    public Location(Parcel in) {
        this.latitude = in.readDouble();
        this.longitude = in.readDouble();
    }

    public static final Parcelable.Creator<Location> CREATOR = new Parcelable.Creator<Location>() {

        public Location createFromParcel(Parcel in) {
            return new Location(in);
        }

        public Location[] newArray(int size) {
            return new Location[size];
        }
    };
    //END PARCELABLE
}

File: MagnetServer/src/Magnet/Model/Location.php
<?php

namespace Magnet\Model;

use \JsonSerializable;

class Location implements JsonSerializable {
	private $latitude;
	private $longitude;

	public function __construct($data = array()) {
		if(is_array($data)) {
			if(isset($data['latitude'])) {
				$this->setLatitude($data['latitude']);
			}

			if(isset($data['longitude'])) {
				$this->setLongitude($data['longitude']);
			}
		}
	}

	public function getLatitude() {
		return $this->latitude;
	}

	public function setLatitude($latitude) {
		if(is_numeric($latitude)) {
			$this->latitude = $latitude;
		}
	}

	public function getLongitude() {
		return $this->longitude;
	}

	public function setLongitude($longitude) {
		if(is_numeric($longitude)) {
			$this->longitude = $longitude;
		}
	}

	public function jsonSerialize() {
		return [
			'latitude' => $this->getLatitude(),
			'longitude' => $this->getLongitude(),
		];
	}
}

?>